### PR TITLE
feat(installations): Firebase JS SDK V9 modular API

### PR DESCRIPTION
**Pull request by exaby73**
### Description

Adds the Firebase JS Web V9 modular API to RNFB. 

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [ ] `iOS`
- My change includes tests;
  - [x] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [x] `jest` tests added or updated in `packages/\*\*/__tests__`
- [x] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [x] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

File: packages/installations/e2e/installations.modular.e2e.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ *
+ */
+
+const jwt = require('jsonwebtoken');
+
+const ID_LENGTH = 22;
+const PROJECT_ID = 448618578101; // this is "magic", it's the react-native-firebase-testing project ID
+
+describe('installations() modular', function () {
+  describe('getId()', function () {
```
Comment (Lyokone) [inline packages/installations/e2e/installations.modular.e2e.js:24]: In the modular.e2e.js files, we keep both the modular and  non-modular in the same file. In future steps of the migration, we will probably remove the non-modular file, and have both files make sure that we don't forget to migrate a test.

File: tests/e2e/.mocharc.js [hunk 2]
```diff
@@ -8,7 +8,6 @@ module.exports = {
   retries: 4,
   bail: true,
   exit: true,
-  recursive: true,
```
Comment (Lyokone) [inline tests/e2e/.mocharc.js:11]: Should we keep this?

Comment (exaby73) [reply tests/e2e/.mocharc.js:11]: It was a duplicate key I removed :) Line 4 already has the same key

File: packages/installations/e2e/installations.modular.e2e.js (hunk 1 above)
Comment (exaby73) [reply packages/installations/e2e/installations.modular.e2e.js]: Added v8 compatibility tests :D

File: tests/e2e/.mocharc.js (hunk 2 above)
Comment (Lyokone) [reply tests/e2e/.mocharc.js:11]: Nice :)

**Review by Lyokone — APPROVED** [review]

**Review by russellwheatley — APPROVED** [review]

**Pull request merged**